server/src/instant/runtime/magic_code_auth.clj

```diff
@@ -58,24 +58,31 @@
 (comment
   (template-replace "Hello {name}, your code is {code}" {:name "Stepan" :code "123"}))
 
-(defn create! [{:keys [app-id email]}]
-  (let [existing-u (app-user-model/get-by-email {:app-id app-id :email email})
-        {user-id :id :as u} (or existing-u
-                                (next-jdbc/with-transaction [conn (aurora/conn-pool :write)]
-                                  (let [app-user (app-user-model/create! conn {:id (random-uuid)
-                                                                               :app-id app-id
-                                                                               :email email})]
-                                    (transaction-model/create! conn {:app-id app-id})
-                                    app-user)))
+(defn create! [{:keys [app-id email user-id]}]
```

Comment:
We could get into trouble if we let users claim an anonymous account with just the user id. Devs often create a public `profile` namespace that shares the id with the user id. An attacker could list profiles and try to claim any new ids they find.

One idea would be to store a random secret locally and store a hash of the secret on the user (or on a new linked `$anonUserProof` system ns). Then when you claim the user, you send the secret and we check that the hash matches.